pkg/authorization/service.go

```diff
@@ -470,6 +546,12 @@ func (a *AuthService) SetupCanvasRoles(canvasID string) error {
 		}
 	}
 
+	// Set up metadata for default canvas roles
+	if err := a.setupDefaultCanvasRoleMetadata(canvasID); err != nil {
+		log.Errorf("Error setting up default canvas role metadata: %v", err)
+		// Don't fail the entire setup if metadata fails
```

Comment:
Why not fail the entire setup if the metadata fails? Isn't the metadata required for this to work properly?